espree.js

```diff
@@ -58,271 +58,38 @@
-
-        // additional adjustment needed if ${ is the last token
-        var terminalDollarBraceL = this.input.slice(result.end, result.end + 2) === "${";
-
-        if (result.range) {
-            result.range[0]--;
-            result.range[1] += (terminalDollarBraceL ? 2 : 1);
-        }
-
-        if (result.loc) {
-            result.loc.start.column--;
-            result.loc.end.column += (terminalDollarBraceL ? 2 : 1);
-        }
-    }
-
-    if (extra.attachComment) {
-        commentAttachment.processComment(result);
-    }
-
-    if (result.type.indexOf("Function") > -1 && !result.generator) {
-        result.generator = false;
+const acorn = require("acorn");
+const jsx = require("acorn-jsx");
+const astNodeTypes = require("./lib/ast-node-types");
+const espree = require("./lib/espree");
+
+// To initialize lazily.
+const parsers = {
+    _regular: null,
+    _jsx: null,
+
+    get regular() {
+        if (this._regular === null) {
+            this._regular = acorn.Parser.extend(espree());
+        }
+        return this._regular;
+    },
+
+    get jsx() {
+        if (this._jsx === null) {
+            this._jsx = acorn.Parser.extend(jsx(), espree());
+        }
+        return this._jsx;
+    },
+
+    get(options) {
+        const useJsx = Boolean(
+            options &&
+            options.ecmaFeatures &&
+            options.ecmaFeatures.jsx
+        );
+        return useJsx ? this.jsx : this.regular;
     }
-
-    return result;
-}
-
-/**
- * Determines if a token is valid given the set of ecmaFeatures.
- * @param {acorn.Parser} parser The parser to check.
- * @returns {boolean} True if the token is allowed, false if not.
- * @private
- */
-function isValidToken(parser) {
-    var ecma = extra.ecmaFeatures;
-    var type = parser.type;
-
-    switch (type) {
-        case tt.jsxName:
-        case tt.jsxText:
-        case tt.jsxTagStart:
-        case tt.jsxTagEnd:
-            return ecma.jsx;
-
-        // https://github.com/ternjs/acorn/issues/363
-        case tt.regexp:
-            if (extra.ecmaVersion < 6 && parser.value.flags && parser.value.flags.indexOf("y") > -1) {
-                return false;
-            }
-
-            return true;
-
-        default:
-            return true;
-    }
-}
-
-/**
- * Injects esprimaFinishNode into the finishNode process.
- * @param {Function} finishNode Original finishNode function.
- * @returns {ASTNode} The finished node.
- * @private
- */
-function wrapFinishNode(finishNode) {
-    return /** @this acorn.Parser */ function(node, type, pos, loc) {
-        var result = finishNode.call(this, node, type, pos, loc);
-        return esprimaFinishNode.call(this, result);
-    };
-}
-
-acorn.plugins.espree = function(instance) {
-
-    instance.extend("finishNode", wrapFinishNode);
-
-    instance.extend("finishNodeAt", wrapFinishNode);
-
-    instance.extend("next", function(next) {
-        return /** @this acorn.Parser */ function() {
```

Comment:
Is this behavior already taken care of in the super class's method?

Reply:
Yes. This override exists to address https://github.com/ternjs/acorn/issues/363 and it has been fixed.